weld/subsystem/src/main/java/org/jboss/as/weld/WeldTransformers.java

```diff
@@ -27,9 +28,9 @@ public void registerTransformers(SubsystemTransformerRegistration subsystem) {
         ResourceTransformationDescriptionBuilder builder400 = chainedBuilder.createBuilder(subsystem.getCurrentSubsystemVersion(), version4_0_0);
         builder400.getAttributeBuilder()
                 // Reject if false or undefined, discard otherwise
-                .addRejectCheck(RejectAttributeChecker.UNDEFINED, WeldResourceDefinition.LEGACY_EMPTY_BEANS_XML_TREATMENT_ATTRIBUTE)
-                .addRejectCheck(new RejectAttributeChecker.SimpleAcceptAttributeChecker(WeldResourceDefinition.LEGACY_EMPTY_BEANS_XML_TREATMENT_ATTRIBUTE.getDefaultValue()), WeldResourceDefinition.LEGACY_EMPTY_BEANS_XML_TREATMENT_ATTRIBUTE)
-                .setDiscard(DiscardAttributeChecker.ALWAYS, WeldResourceDefinition.LEGACY_EMPTY_BEANS_XML_TREATMENT_ATTRIBUTE)
+                .setDiscard(new DiscardAttributeChecker.DiscardAttributeValueChecker(false, true, ModelNode.TRUE),
```

Comment:
Maybe I'm misreading this but doesn't this discard the config if its undefined as well? I thought we wanted to reject it in such case.

Reply:
Oh, crud; you are right! I mixed up the params. I merged your branch; I'll send up a PR to fix this.

Reply:
Cool, thanks!